test/models/trip_test.dart

```diff
@@ -0,0 +1,369 @@
+          startTime: DateTime.now().add(const Duration(hours: 2)),
+          endTime: DateTime.now().add(const Duration(hours: 4)));
+      expect(trip.overlapsWithTimeRange(range), true);
+    });
+  });
+  group('Trip.shouldShowInListView', () {
+    test('returns true if trip should show in ListView', () {
+      final Trip ride = RideFactory().generateFake(
+        status: RideStatus.approved,
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(ride.shouldShowInListView(past: false), true);
+      final Trip drive = DriveFactory().generateFake(
+        cancelled: false,
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(drive.shouldShowInListView(past: false), true);
+    });
+    test('returns false for trip with hideInListView', () {
+      final Trip ride = RideFactory().generateFake(
+        hideInListView: true,
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(ride.shouldShowInListView(past: false), false);
+      final Trip drive = DriveFactory().generateFake(
+        hideInListView: true,
+        startTime: DateTime.now().add(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(drive.shouldShowInListView(past: false), false);
+    });
+    test('returns true for past finsihed trip', () {
+      final Trip ride = RideFactory().generateFake(
+        status: RideStatus.approved,
+        startTime: DateTime.now().subtract(const Duration(hours: 4)),
+        endTime: DateTime.now().subtract(const Duration(hours: 2)),
+      );
+      expect(ride.shouldShowInListView(past: true), true);
+      final Trip drive = DriveFactory().generateFake(
+        cancelled: false,
+        startTime: DateTime.now().subtract(const Duration(hours: 4)),
+        endTime: DateTime.now().subtract(const Duration(hours: 2)),
+      );
+      expect(drive.shouldShowInListView(past: true), true);
+    });
+    test('returns false in ongoing trip', () {
+      final Trip ride = RideFactory().generateFake(
+        status: RideStatus.pending,
+        startTime: DateTime.now().subtract(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(ride.shouldShowInListView(past: true), false);
+      final Trip drive = DriveFactory().generateFake(
+        cancelled: true,
+        startTime: DateTime.now().subtract(const Duration(hours: 2)),
+        endTime: DateTime.now().add(const Duration(hours: 4)),
+      );
+      expect(drive.shouldShowInListView(past: true), false);
+    });
+  });
+  group('Trip.shouldShowInListView', () {
+    test('shows in ListView', () {
```

Comment:
This kind of sounds like a widget test. I would try to go for something more descriptive of the test case, like "future trip in future list view" or something

Reply:
Also, what is the difference between this and the `Trip.shouldShowInListView` group before?

Reply:
forgot to remove the old one